web/src/components/ErrorScreen.js

```diff
@@ -17,106 +15,126 @@ export default class ErrorScreen extends React.Component {
             <h1>{props.header}</h1>
           </header>
           <div className="scroller-wrap">
-            <div className="details scroller">{props.body}</div>
+            <div className="details scroller">{props.children}</div>
           </div>
           <footer>{props.footer}</footer>
         </div>
       </div>
+    </div>
+  );
+};
+
+export default class ErrorScreenWrapper extends React.Component {
+  renderUnknownError() {
+    return (
+      <ErrorScreen header="Wha?!">
+        <section>
+          <p>???</p>
+        </section>
+      </ErrorScreen>
     );
   }
 
-  makeInner() {
-    const {type, data} = this.props;
+  renderNoAuth() {
+    return (
+      <ErrorScreen header="Huh?!">
+        <section>
+          <p>Well, this is embarrassing. I don't know how you got here!</p>
+          <p>This isn't normal at all, so I'm not really sure how to proceed...</p>
+          <p>Maybe try refreshing? Let's hope for the best!</p>
+        </section>
+      </ErrorScreen>
+    );
+  }
+
+  renderDisconnected() {
+    return (
+      <ErrorScreen header="Disconnected!">
+        <section>
+          <p>Sorry! I might have gone to sleep for the night!</p>
+          <p>If you know I'm supposed to be awake, try giving this page a refresh!</p>
+        </section>
+      </ErrorScreen>
+    );
+  }
+
+  renderNoAccess() {
+    const errorFooter = (
+      <form method="POST" action="logout">
+        <button className="primary" type="submit">
+          Re-authenticate
+        </button>
+      </form>
+    );
+    return (
+      <ErrorScreen header="Access denied!" footer={errorFooter}>
+        <div>
+          <section>
+            You need to be on:
+            <Guild />
+            {/* TODO */}
+          </section>
+          <section>
+            You are logged in as:
+            <div className="member">
+              <UserAvatar user={this.props.data.user.id} avatar={this.props.data.user.avatar} />

```

Comment:
oh wow this is gross, we should destructure these

you don't have to do it in this pr; if you decide not to, lmk and i'll patch it up myself after merge

Reply:
Yeah it's a pretty hard to read mess. I'd prefer not to destructure it (right now) but if you want I can help do it it later?